### DISCUSSION TITLE
RabbitMQ Channel Closes when debugging

### DISCUSSION THREAD

**Discussion by jonatassallazar** [golevelup/nestjs#709] [Q&A]
Hello!

We use the @golevelup/nestjs-rabbitmq package in our app, and it works perfectly to us, but when we need to debug the application to find and kill the bugs, just after the code stops at the breakpoint and we move some lines or letting the app run again, the rabbitmq throw an error of "Disconnected from RMQ. Trying to reconnect" and shortly after "Unexpected close", then I have to restart the app to work again. 

Anyone has the same problem? I've tried some configurations in the rabbitmq options, but no luck until now. 

I really don't know if it's really the default behaviour or if the setup is something wrong or even if it's some config on rabbitmq container connection options.

I aprettiate any help or ideas.

Thanks in advance.

**Comment by bladerunner2020** [comment]
Though this question was asked a while ago, I think the answer could still be useful to others. After struggling with the same issue, we came up with the following solution:

```ts
@Module({
  imports: [
    ConfigModule,
    RabbitMQModule.forRootAsync({
      inject: [ConfigService],
      useFactory: (configService: ConfigService) => {
        const RMQ_HOST = configService.get('RMQ_HOST') || 'localhost';
        const RMQ_PORT = configService.get('RMQ_PORT') || 5672;
        const RMQ_USER = configService.get('RMQ_USER');
        const RMQ_PASS = configService.get('RMQ_PASS');

        const uri = RMQ_USER && RMQ_PASS
          ? `amqp://${RMQ_USER}:${RMQ_PASS}@${RMQ_HOST}:${RMQ_PORT}`
          : `amqp://${RMQ_HOST}:${RMQ_PORT}`;

        return {
          uri,
          exchanges: [ ... ],
          connectionManagerOptions: {
            // Add this line
            heartbeatIntervalInSeconds: process.env.NODE_ENV === 'development'
                ? DEBUG_HEARTBEAT_INTERVAL_SECONDS
                : undefined,
          },
          connectionInitOptions: {
            wait: false,
            reject: false,
          },
        };
      },
    }),
    RmqModule,
  ],
```

It doesn't completely solve the issue, but it definitely makes life much easier.
